job-runner/chain/reaper.go

```diff
@@ -0,0 +1,435 @@
+// Copyright 2018, Square, Inc.
+
+package chain
+
+import (
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/square/spincycle/proto"
+	rm "github.com/square/spincycle/request-manager"
+	"github.com/square/spincycle/retry"
+)
+
+const (
+	// Number of times to attempt sending chain state / SJC to RM.
+	FINALIZE_TRIES = 5
+	// Time to wait between the above tries.
+	FINALIZE_RETRY_WAIT = 500 * time.Millisecond
+)
+
+// A JobReaper handles jobs and chains that have finished running.
+type JobReaper interface {
```

Comment:
Let's document more here: why we have varying implementations of reapers. Or document it elsewhere. It's an important design decision.